File: sol/1394/1394.cpp
struct Solution {
    int findLucky(vector<int>& arr) {
        int n = arr.size();
        vector<int> freq(n + 1, 0);
        for (int x : arr){
            if (x <= n){
                ++freq[x];
            }
        }
        for (int i = n; i > 0; --i){
            if (freq[i] == i){
                return i;
            }
        }
        return -1;
    }
};

File: sol/1394/1394.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        freq = dict()
        for i in arr:
            freq[i] = freq.get(i, 0) + 1
        res = -1
        for k, v in freq.items():
            if k == v:
                res = max(res, k)
        return res
